lib/ansible/playbook/role/__init__.py

```diff
@@ -144,6 +152,10 @@ def __init__(self, play=None, from_files=None, from_include=False, validate=True
     def __repr__(self):
         return self.get_name()
 
+    @cached_property
+    def full_qualified_name(self):
```

Comment:
I think it's more common to say "fully-qualified"
```suggestion
    def fully_qualified_name(self):
```